includes/class-amp-post-type-support.php

```diff
@@ -49,6 +49,28 @@ public static function get_eligible_post_types() {
 		);
 	}
 
+	/**
+	 * Get post types that can be shown in the REST API and supports AMP.
+	 *
+	 * @return string[] Post types.
+	 */
+	public static function get_post_types_for_rest_api() {
+		if ( amp_is_canonical() ) {
+			$post_types = get_post_types_by_support( 'editor' ); // @todo Shouldn't this actually only be those with 'amp' support, or if if all_templates_supported?
```

Comment:
@westonruter Does this `todo` still apply?

Reply:
Yes, I think so. Some old code that needs to be re-evaluated.

Reply:
Actually, it seems to be entirely incorrect. Consider the case where you're using Standard mode but disable AMP for the pages post type:

![image](https://user-images.githubusercontent.com/134745/82157526-d6587a00-9836-11ea-94ea-5143d8b8a979.png)

The REST API currently then for pages is still including these fields:

![image](https://user-images.githubusercontent.com/134745/82157585-2c2d2200-9837-11ea-9a4c-27ce7269e6f6.png)

This is wrong. They should be omitted entirely. They don't need to be there.

I think this condition was put in place at a time where I thought that canonical should imply that AMP cannot be disabled for any part of the site. But this is not correct.

Reply:
The method `AMP_Post_Type_Support::add_post_type_support()` will ensure that all post types have `amp` post type supported added when the “serve all templates” checkbox is checked.